atc/db/component.go

```diff
@@ -56,8 +57,17 @@ func (c *component) Reload() (bool, error) {
 	return true, nil
 }
 
+// IntervalElapsed adds a random tiny drift [-1, 1] seconds to interval, when
+// there are multiple ATCs, each ATC uses a slightly different interval, so
+// that shorter interval gets more chance to run. This mechanism helps distribute
+// component to across ATCs more evenly.
 func (c *component) IntervalElapsed() bool {
-	return time.Now().After(c.lastRan.Add(c.interval))
+	interval := c.interval
+	drift := time.Duration(rand.Int())%(2*time.Second) - time.Second
```

Comment:
Maybe it should seed the rand to make it truely random?

Reply:
Without a seed, when each ATC starts, they will all start with the same random number. But in this case, I think that is ok, because we just want a random drift. Our prod cluster's metrics prove the current code is working fine.

But to honor your comment, I added a seed.